test/e2e/autotls/auto_tls_test.go

```diff
@@ -115,13 +119,86 @@ func httpsReady(svc *servingv1.Service) (bool, error) {
 	}
 }
 
-func createRootCAs(t *testing.T, clients *test.Clients, ns, secretName string) *x509.CertPool {
+func TestRouteModified(t *testing.T) {
+	clients := e2e.Setup(t)
+	disableNamespaceCertWithWhiteList(t, clients, sets.String{})
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "runtime",
+	}
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	defer test.TearDown(clients, names)
+
+	objects, err := v1test.CreateServiceReady(t, clients, &names)
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
+
+	secretName := routenames.Certificate(objects.Route)
+	rootCAs := createRootCAs(t, clients, objects.Route.Namespace, secretName)
+
+	var transportOption spoof.TransportOption = func(transport *http.Transport) *http.Transport {
+		transport.TLSClientConfig = &tls.Config{RootCAs: rootCAs}
+		return transport
+	}
+	prober := test.RunRouteProber(t.Logf, clients, objects.Service.Status.URL.URL(), transportOption)
```

Comment:
I am not sure if we could add this prober into the TestAutoTLS test with your PR https://github.com/knative/serving/pull/6890 so we don't need the TestRouteModified test as seems like it retested some stuffs (e.g. the HTTPS to the main Route).